flink-table-store-e2e-tests/src/test/java/org/apache/flink/table/store/tests/E2eTestBase.java

```diff
@@ -55,95 +51,70 @@ public abstract class E2eTestBase {
 
     private static final Logger LOG = LoggerFactory.getLogger(E2eTestBase.class);
 
-    // ------------------------------------------------------------------------------------------
-    // Flink Variables
-    // ------------------------------------------------------------------------------------------
-    private static final String FLINK_IMAGE_TAG;
-
-    static {
-        Properties properties = new Properties();
-        try {
-            properties.load(
-                    E2eTestBase.class.getClassLoader().getResourceAsStream("project.properties"));
-        } catch (IOException e) {
-            throw new RuntimeException(e);
-        }
-        // TODO change image tag to official flink image after 1.15 is released
-        FLINK_IMAGE_TAG = "tsreaper/flink-test:" + properties.getProperty("flink.version");
+    private final boolean withKafka;
+
+    protected E2eTestBase() {
+        this(false);
+    }
+
+    protected E2eTestBase(boolean withKafka) {
+        this.withKafka = withKafka;
     }
 
-    private static final String INTER_CONTAINER_JM_ALIAS = "jobmanager";
-    private static final String INTER_CONTAINER_TM_ALIAS = "taskmanager";
-    private static final int JOB_MANAGER_REST_PORT = 8081;
-    private static final String FLINK_PROPERTIES =
-            String.join(
-                    "\n",
-                    Arrays.asList(
-                            "jobmanager.rpc.address: jobmanager",
-                            "taskmanager.numberOfTaskSlots: 9",
-                            "parallelism.default: 3",
-                            "sql-client.execution.result-mode: TABLEAU"));
-
-    // ------------------------------------------------------------------------------------------
-    // Additional Jars
-    // ------------------------------------------------------------------------------------------
     private static final String TABLE_STORE_JAR_NAME = "flink-table-store.jar";
     private static final String BUNDLED_HADOOP_JAR_NAME = "bundled-hadoop.jar";
-
-    protected static final String TEST_DATA_DIR = "/opt/flink/test-data";
-
-    @ClassRule public static final Network NETWORK = Network.newNetwork();
-    private GenericContainer<?> jobManager;
-    protected GenericContainer<?> taskManager;
+    protected static final String TEST_DATA_DIR = "/test-data";
 
     private static final String PRINT_SINK_IDENTIFIER = "table-store-e2e-result";
     private static final int CHECK_RESULT_INTERVAL_MS = 1000;
     private static final int CHECK_RESULT_RETRIES = 60;
     private final List<String> currentResults = new ArrayList<>();
 
+    private DockerComposeContainer<?> environment;
+    private static final Object environmentStartLock = new Object();
```

Comment:
I'm a little curious as to why checkstyle didn't fail, it looks like the name `environmentStartLock` should be upper here.
Maybe we can just `synchronized(E2eTestBase.class)`